lib/Runtime/Library/BuiltIn/IndexOf.js

```diff
@@ -0,0 +1,48 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+"use strict";
+
+(function indexOf(searchElement, fromIndex) {
+
+    var k;
+
+    if (this == null) {
+        throw new TypeError('"this" is null or not defined');
+    }
+
+    var o = this;
+
+    var len = o.length >>> 0;
```

Comment:
Just a note, from testing it seems that no annotation is fine here, `|0` would be wrong, and `>>>0` is correct but not strictly necessary, except for potentially providing type information.